src/engine/enginebuffer.cpp

```diff
@@ -538,7 +541,24 @@ void EngineBuffer::slotTrackLoaded(TrackPointer pTrack,
     m_visualPlayPos->setInvalid();
     m_playPos = kInitialPlayPosition; // for execute seeks to 0.0
     m_pCurrentTrack = pTrack;
-    m_pTrackSamples->set(trackNumSamples);
+
+    m_channelCount = trackChannelCount;
+    if (m_channelCount > mixxx::audio::ChannelCount::stereo()) {
+        // The sample count is indicated downmix. This means that for stem
+        // track, we only consider the track in stereo, as it is perceived by
+        // the user on deck output
+        VERIFY_OR_DEBUG_ASSERT(m_channelCount % mixxx::audio::ChannelCount::stereo() == 0) {
+            // Make it stereo for the frame calculation
+            kLogger.warning() << "Odd number of channel in the track is not supported";
+        };
+    } else {
+        // The EngineBuffer only works with stereo channels. If the track is
```

Comment:
This cannot be asserted at that level as the `AudioSourceStereoProxy` will be instantiated by `CachingReaderChunk` and won't return a channel count to the `EngineBuffer`. Do you have a suggestion on how this could be asserted otherwise?

Reply:
OK, than keep it as it is.